lightning-invoice/src/utils.rs

```diff
@@ -286,72 +286,45 @@ where
 	)
 }
 
-/// See [`create_invoice_from_channelmanager_with_description_hash`]
-/// This version can be used in a `no_std` environment, where [`std::time::SystemTime`] is not
-/// available and the current time is supplied by the caller.
-pub fn create_invoice_from_channelmanager_with_description_hash_and_duration_since_epoch<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
-	channelmanager: &ChannelManager<M, T, K, F, L>, keys_manager: K, logger: L,
-	network: Currency, amt_msat: Option<u64>, description_hash: Sha256,
-	duration_since_epoch: Duration, invoice_expiry_delta_secs: u32
-) -> Result<Invoice, SignOrCreationError<()>>
-where
-	M::Target: chain::Watch<<K::Target as KeysInterface>::Signer>,
-	T::Target: BroadcasterInterface,
-	K::Target: KeysInterface,
-	F::Target: FeeEstimator,
-	L::Target: Logger,
-{
-	_create_invoice_from_channelmanager_and_duration_since_epoch(
-		channelmanager, keys_manager, logger, network, amt_msat,
-		InvoiceDescription::Hash(&description_hash),
-		duration_since_epoch, invoice_expiry_delta_secs
-	)
-}
-
-/// See [`create_invoice_from_channelmanager`]
-/// This version can be used in a `no_std` environment, where [`std::time::SystemTime`] is not
-/// available and the current time is supplied by the caller.
-pub fn create_invoice_from_channelmanager_and_duration_since_epoch<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
+///Function that allows to construct invoices with the given payment hash.
```

Comment:
This one is difficult for me because I am not completely sure why a user would want a manual payment hash either. I also wasn't quite understanding the formats of the other functions descriptions, so was pretty unsure what the best description would be. Help here would be appreciated.

Reply:
ok, I hope what I did was good, and I hopefully reduced the diff for easier review.